### PR TITLE
feat: create cosmos_node_client package

### LINKED ISSUE #255: [Feature]  Create 'cosmos_node_client' package
Create new package called `cosmos_node_client` for communicating with the Cosmos nodes.

## Acceptance criteria

- new package structure inside the `packages` subfolder without any code yet with `comsos_lint` #package plugged in for static code analysis
- amend the `pull-request.yaml` file to perform all the necessary checks on the new package

### PR DESCRIPTION
**Pull request by Zfinix**
Closes #255

### CODE REVIEW

**Comment by Zfinix** [comment]
This is ready guys

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -66,6 +67,30 @@ jobs:
...
+
+      - name: cosmos_node_client - dart analysis
+        working-directory: packages/cosmos_node_client
+        if: env.cosmos_node_client == 'true'
+        run: flutter analyze
+
+     # - name: cosmos_node_client - tests
+        #working-directory: packages/cosmos_node_client
+        #if: env.cosmos_node_client == 'true'
+        #run: flutter test
```
Comment (andrzejchm) [inline .github/workflows/pull-request.yml:93]: any reason this is commented out?

File: packages/cosmos_node_client/analysis_options.yaml [hunk 2]
```diff
@@ -0,0 +1,228 @@
+include: package:lint/analysis_options.yaml
```
Comment (andrzejchm) [inline packages/cosmos_node_client/analysis_options.yaml:1]: let's use `cosmos_lint` package here instead of specifying the rules again

**Review by andrzejchm — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (Zfinix) [reply .github/workflows/pull-request.yml:93]: We don't have any tears right now and the ci fails because of that

Comment (andrzejchm) [reply .github/workflows/pull-request.yml:93]: lets add one, even 2+2=4 test just to enable the CI from the beginning. you can always remove the test as soon as you add proper ones later on

File: packages/cosmos_node_client/analysis_options.yaml (hunk 2 above)
Comment (Zfinix) [reply packages/cosmos_node_client/analysis_options.yaml:1]: Yeah I'm going to do that just needed the previous PR to be merged.

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (Zfinix) [reply .github/workflows/pull-request.yml:93]: Okay

**Review by andrzejchm — APPROVED** [review]
👏🏻

**Pull request merged**